build-support/travis/travis.yml.mustache

```diff
@@ -105,16 +109,20 @@ pants_run_cache_config: &pants_run_cache_config
 # we directly clone the Pyenv repo.
 
 pyenv_setup: &pyenv_setup >
-  if [ ! -d "${PYENV_BIN}" ]; then
-  rm -rf "${PYENV_ROOT}";
-  git clone https://github.com/pyenv/pyenv "${PYENV_ROOT}";
+  if [[ ! -x "${PYENV_BIN}" ]]; then
+    rm -rf "${PYENV_ROOT}"
+    git clone https://github.com/pyenv/pyenv "${PYENV_ROOT}"
   fi
 
 pyenv_install_py36: &pyenv_install_py36 >
-  "${PYENV_BIN}" install "${PYENV_PY36_VERSION}"
+  if [[ ! -d ${PYENV_ROOT}/versions/"${PYENV_PY36_VERSION}" ]]; then
+    "${PYENV_BIN}" install "${PYENV_PY36_VERSION}"
+  fi
 
```

Comment:
Is it possible to make this boilerplate into a mustache template?

Reply:
We can't make it a Mustache template because we need to parameterize the Python version to install, and the only way to do that is through env vars in Mustache, which wouldn't allow us to do something like 

```yml
script:
- * pyenv_install_py36
- * pyenv_install_py37
```

in the same shard.

--

I can make this a simple script if you'd like! Similar to https://github.com/pantsbuild/pants/blob/master/build-support/bin/install_aws_cli_for_ci.sh.

The script can setup Pyenv if necessary, allowing us to remove that entry, and then take a parameter for which version(s) to install.

Reply:
I would really like to make this into a script -- taking conditional and script logic out of the travis config feels much easier to review and expand on. I would recommend a script which checks whether pyenv is already installed, clones it if not, then installs all the required versions for the given shard (provided as positional arguments) -- or something like that, maybe.